python_fundamentals-master/08_file_io/08_01_words_analysis.py

```diff
@@ -7,3 +7,27 @@
 
 
 '''
+with open("words.txt" , "r+") as f:
+    word_min = "hello"
+    word_max = ""
+    word_max_len = 0
+    words_min = []
+    words_max = []
+    total_num_words = 0
+
+    for word in f:
+        if len(word) <= len(word_min):
+            word_min = word
+            words_min.append(word_min.strip())
+        if len(word) == word_max_len:
+            words_max.append(word)
+        elif len(word) > word_max_len:
+            words_max = [word]
+            word_max_len = len(word)
+        total_num_words += 1
+
+    print(words_min)
+    print(words_max)
+    print(total_num_words)
+
+#Debugger is great :D
```

Comment:
Oh ok, so this got worked on more since the previous one. Glad you are using the debugger!

So, I still see a few potential problems here.  The same problem as before I think, for word_min, it will give me all the minimum words that were found on the way.

For word_max you are resetting it to be a single word if it finds a new one. This is smart and probably what you need to do for words_min.  This ensures that if a word is larger (and not tying) it will be alone at the end.

You probably just didn't have this case for word-min, but it could certainly happen as well.

Reply:
I don't understand what you want to see. I want all the words printed with two letters right? "if it is a tie, print all"

Reply:
Yes, but what if I find these words in this order?
'hello' , 'high', 'man', 'to', 'ad', 'ba'
Then it should only print `to, ad, ba` but I think it will currently print all of them.

I ran this code
```python
f = ["hello", "high", "bab", "ba", "da", "to"]

word_min = "hello"
word_max = ""
word_max_len = 0
words_min = []
words_max = []
total_num_words = 0

for word in f:
    if len(word) <= len(word_min):
        word_min = word
        words_min.append(word_min.strip())
    if len(word) == word_max_len:
        words_max.append(word)
    elif len(word) > word_max_len:
        words_max = [word]
        word_max_len = len(word)
    total_num_words += 1

print(f"words min : {words_min}")
print(f"words_max : {words_max}")
print(total_num_words)```

and it gave me this output

```python
words min : ['hello', 'high', 'bab', 'ba', 'da', 'to']
words_max : ['hello']
6
```

That's not what you want.